### PR TITLE
Solution to the issue #2741

### PR DESCRIPTION
**Pull request by brazdilj**
Issue: #2741
https://github.com/MoneyFox/MoneyFox/issues/2741

## PR Type
What kind of change does this PR introduce?

- Feature

## What is the current behavior?
Currently the number of categories included in the category spreading is fix. It would be nice if this could be dynamically selected.
This should be persisted to the settings.

## What is the new behavior?
Number of categories included in the category spreading can be changed dynamically via entry or increase/decrease buttons.
It is also persisted to the settings.

## PR Checklist

Not tested on iOS (Unfortunately, I do not own any Apple product)

- [x] Tested code on Windows
- [x] Tested code on Android
- [ ] Tested code on iOS
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes

## Other information

Entry is cleared for edge cases as well as any non numeric input.

Number of categories is persisted via SettingsFacade.cs just like in AccountGroup.cs, but I am not so sure with my solution so maybe it needs a correction.

Also text in the new label for the entry is hardcoded in English, I did not add all the necessary translations which I apologize for.

### CODE REVIEW

File: Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml [hunk 1]
```diff
@@ -36,15 +36,47 @@
...
 
+                <Grid Grid.Row="2" ColumnDefinitions="*, Auto" Padding="0,15,0,0" VerticalOptions="Start">
+                    <Label Style="{StaticResource TextBodySecondary}"
+                           Text="Number of categories to show:"
+                           VerticalOptions="Center"/>
+
+                    <FlexLayout Grid.Column="1"
+                                Direction="Row"
+                                JustifyContent="SpaceBetween">
+                        <Button Text="-"
```
Comment (NPadrutt) [inline Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml:54]: Couldn't you just use a Stepper here? I think that would simplify the code in the viewmodel as well.

**Review by NPadrutt — CHANGES_REQUESTED** [review]

File: Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml (hunk 1 above)
Comment (brazdilj) [reply Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml:54]: I could, I was actually thinking about using Stepper, but I wanted to have the Entry between the two buttons. Is it possible with Stepper?

Comment (NPadrutt) [reply Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml:54]: Mh, I don't think so. But then I would create your own little steper component and use that. Then you don't have to have Increment and Decrement commands in the  viewmodel of that concrete page, but can hide those in the component and just make a ValueChange Event.

Comment (brazdilj) [reply Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml:54]: You're right. So I'll create a custom component. I'll let you know when it's ready.

Comment (brazdilj) [reply Src/MoneyFox.Ui/Views/Statistics/CategorySpreading/StatisticCategorySpreadingPage.xaml]: Hello, so I created a custom component called CustomStepperWithEntry and moved a lot of code from the viewmodel to its codebehind.

**Review by NPadrutt — APPROVED** [review]

**Comment by NPadrutt** [comment]
Thank you for your PR @brazdilj  :)

**Review by NPadrutt — APPROVED** [review]

**Review by NPadrutt — APPROVED** [review]

**Comment by brazdilj** [comment]
You're welcome @NPadrutt and thank you too for your help :)
